app/controllers/search_controller.rb

```diff
@@ -0,0 +1,40 @@
+class SearchController < ApplicationController
+
+  def search
+    @search = Search.new
+    @search.search_fields.build
+  end
+
+  def show
+    @search = Search.new(search_params)
+    if(@search.search_fields.blank?)
+      @search.search_fields.build
+    end
+    query_string = make_query params
+    @papers = Paper.where(query_string)
+  end
+
+
+
+  private
+
+  def search_params
+    params.require(:search).permit(:name, :saved_on,
+                                   :search_fields_attributes => [:content,:field,:op1,:op2,:_destroy]);
+  end
+
+  #make query string for PG
+  def make_query params
```

Comment:
We have another solution for search. We could consider it in next sprint, because it could help us sort result and search in association such as `has_many` and `belongs_to`.

[ransack](https://github.com/activerecord-hackery/ransack)